File: lsp/file/basic/file_open.c
#include <stdio.h>
#include <string.h>

int write_to_file(void)
{
    FILE *fp; // file pointer

    // Truncate file to zero length or create text file for writing
    fp = fopen("data", "w");
    if (fp == NULL) {
        perror("fopen error\n");
        return -1;
    }
    // write to file
    fputs("hello world..\n", fp);
    fclose(fp);

    return 0;
}

int read_from_file(void)
{
    FILE *fp; // file pointer
    char buf[1024];

    // Open text file for reading
    fp = fopen("data", "r");
    if (fp == NULL) {
        perror("fopen error\n");
        return -1;
    }
    // read file
    memset(buf, 0, sizeof(buf));
    fgets(buf, sizeof(buf), fp);
    fclose(fp);

    printf("%s\n", buf);

    return 0;
}

int main(int argc, char **argv)
{
    if (write_to_file()) {
        perror("write to file\n");
        return -1;
    }

    if (read_from_file()) {
        perror("read from file\n");
        return -1;
    }

    return 0;
}

File: lsp/file/basic/file_open.go
package fileopen

import (
	"fmt"
	"io"
	"os"
)

func WriteToFile(fileName string) {
	// file open
	file, err := os.Create(fileName)
	if err != nil {
		panic(err)
	}
	defer func() {
		if err := file.Close(); err != nil {
			panic(err)
		}
	}()

	n, err := file.WriteString("hello world\n")
	if err != nil {
		panic(err)
	}
	fmt.Printf("wrote %d bytes\n", n)
}

func ReadFromFile(fileName string) {
	// file open
	file, err := os.Open(fileName)
	if err != nil {
		panic(err)
	}
	defer func() {
		if err := file.Close(); err != nil {
			panic(err)
		}
	}()
	buf := make([]byte, 1024)
	for {
		// read chunk
		n, err := file.Read(buf)
		if err != nil && err != io.EOF {
			panic(err)
		}
		if n == 0 {
			break
		}

		fmt.Printf("%d bytes: %s\n", n, string(buf))
	}

}
